packages/block-library/src/image/image-size.js

```diff
@@ -9,6 +9,11 @@ import { noop } from 'lodash';
 import { withGlobalEvents } from '@wordpress/compose';
 import { Component } from '@wordpress/element';
 
+export function getBlockWidth() {
```

Comment:
I've confirmed that depending on whether the image block is Wide/Full aligned and whether it's the first block will return a different value. For example, if I've a wide image block as the sole content, this returns `718`. If I insert a paragraph before the image block, it suddenly starts returning `580` (the width of the non-wide paragraph).

We should assign a [`ref`](https://reactjs.org/docs/refs-and-the-dom.html) to the image from which to inspect the DOM width.

Reply:
> We should assign a [`ref`](https://reactjs.org/docs/refs-and-the-dom.html) to the image from which to inspect the DOM width.

Except: Is it really the image width we care about? Or is the end-goal to get what is effectively the rendered width of the block list (non-wide, non-full)? (Which I assume would ideally be the same as `$content_width`)

Reply:
> Noting that this would return the width value of any block, not necessarily the image block.

Right, this is the intent. See the comment above.

> a block shouldn't assume anything about its ancestry, as it makes for a maintainability nightmare should we ever change the class of the block wrapper.

> Except: Is it really the image width we care about? Or is the end-goal to get what is effectively the rendered width of the block list (non-wide, non-full)? (Which I assume would ideally be the same as $content_width)

The latter. Unfortunately there's no "easy way" to get the editor "visual" width at any given moment. We need to look at the DOM somewhere. Any suggestions on how to improve that are very welcome :)

The old `$content_width`, even if we reuse it again would not work here as themes can set the width from CSS. My idea was to "fix" the width to content_width or a similar new global that would be used as "max-width" by themes and enforced in the editor and the front-end.

It's probably not too late to do that, just need to set it as expectation for the themes.

Reply:
If we needed to reach into the DOM, and it's expected a theme would provide their custom width by overriding the `max-width` style of `.wp-block`, and we have no guarantee there'd be a non-full/wide align block present in the editor, I might imagine there could be an option with creating a DOM node for the sole purpose of inspecting its computed style:

```js
const div = document.createElement( 'div' );
div.className = 'wp-block'
const { maxWidth } = window.getComputedStyle( div );
// '610px'
document.body.removeChild( div );
```

There's a couple problems here:

- It's tragically non-performant if subject to repeated calls
- To get from 610 to 580, we need to subtract the paddings of the block, which vary depending on screen size, and are styled contextualized to `.editor-block-list__layout .editor-block-list__block`

Reply:
> If we needed to reach into the DOM, and it's expected a theme would provide their custom width by overriding the `max-width` style of `.wp-block`, and we have no guarantee there'd be a non-full/wide align block present in the editor, I might imagine there could be an option with creating a DOM node for the sole purpose of inspecting its computed style:

I went with something like this in 87130e4, considering it a temporary solution, cached as much as is reasonable (cleared on window resize).